apps/aecore/src/aec_chain_state.erl

```diff
@@ -106,14 +109,25 @@ top_block(?match_state(top_block_hash := X) = State) ->
     export_block(blocks_db_get(X, State), State).
 
 -spec insert_block(#block{}, state()) -> {'ok', state()} | {'error', any()}.
-insert_block(Block, ?assert_state() = State) ->
-    try {ok, internal_insert(wrap_block(Block), State)}
+insert_block(Block, ?assert_state() = State0) ->
+    try 
+        State1 = internal_insert(wrap_block(Block), State0),
+        State =
+            case Block#block.height of
+                0 ->
+                    {ok, GenesisHash} =
+                        aec_blocks:hash_internal_representation(Block),
+                    State1#{genesis_block_hash => GenesisHash};
+                _ ->
+                    State1
+            end,
+        {ok, State}
     catch throw:?internal_error(What) -> {error, What}
     end.
 
 -spec insert_header(#header{}, state()) -> {'ok', state()} | {'error', any()}.
-insert_header(Header, ?assert_state() = State) ->
-    try {ok, internal_insert(wrap_header(Header), State)}
+insert_header(Header, ?assert_state() = State0) ->
```

Comment:
Either there is some genesis stuff missing, or you should revert to the old code.